python/interpreter_extensions.bzl

```
@@ -0,0 +1,55 @@
+# Copyright 2023 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"Module extensions for use with bzlmod and pip interpreters"
+
+load("@pythons_hub//:interpreters.bzl", "INTERPRETER_LABELS")
+
+def _pypi_ext_impl(mctx):
+    # TODO do we want to run this more than once with multiple
+    # python interpreters?
+    # If so we need to do the for loop
+    install = mctx.modules[0].tags.install[0]
+    _pypi_repo(
+        name = install.name,
+        python_name = install.python_name,
+    )
+
+pypi_ext = module_extension(
+    implementation = _pypi_ext_impl,
+    tag_classes = {
+        "install": tag_class(
+            attrs = {
+                "name": attr.string(),
+                "python_name": attr.string(),
+            },
+        ),
+    },
+)
+
+def _pypi_repo_impl(rctx):
+    rctx.file("BUILD", "exports_files(['python'])")
+
+    if rctx.attr.python_name in INTERPRETER_LABELS:
+        actual_interpreter_label = INTERPRETER_LABELS[rctx.attr.python_name]
+        rctx.symlink(actual_interpreter_label, "python")
+    else:
+        fail("unable to find the interpreter")
```

nit: you could simplify this with the following:
```suggestion
    actual_interpreter_label = INTERPRETER_LABELS.get(rctx.attr.python_name)
    if actual_interpreter_label is None:
        fail("unable to find the interpreter")
    
    rctx.symlink(actual_interpreter_label, "python")
```